File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
arr = list(map(int, input().split()))

start = 0
end = 0
arr_sum = arr[0]
answer = 0
while True:
    if arr_sum < m:
        end += 1
        if end == n:
            break
        arr_sum += arr[end]
    elif arr_sum > m:
        arr_sum -= arr[start]
        start += 1
    else:
        answer += 1
        end += 1
        if end == n:
            break
        arr_sum = arr_sum + arr[end] - arr[start]
        start += 1

print(answer)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const [n, m] = input[0].split(' ').map(Number)
const arr = input[1].split(' ').map(Number)

const sum_arr = [0]
let sum = 0
for (const cur of arr) {
  sum += cur
  sum_arr.push(sum)
}

let answer = 0
let left = 0
let right = 0
while(right < n + 1) {
  const sum = sum_arr[right] - sum_arr[left]
  
  if (m === sum) {
    answer++
    right++
  }
  else if (m < sum) left++
  else if (m > sum) right++
}
console.log(answer)